components/raftstore/src/store/worker/region.rs

```diff
@@ -763,6 +817,14 @@ where
             }
             task @ Task::Apply { .. } => {
                 fail_point!("on_region_worker_apply", true, |_| {});
+                if self.ctx.coprocessor_host.should_pre_apply_snapshot() {
+                    if let Err(e) = self.ctx.pre_apply_snapshot(&task) {
```

Comment:
these pre-apply-snapshots are handled in serial instead of concurrently?

Reply:
They are handled in parallel